module/vision/BallDetector/src/BallDetector.cpp

```diff
@@ -159,14 +160,15 @@ namespace module::vision {
 
                         // Attach the measurement to the object (distance from camera to ball)
                         b.measurements.push_back(Ball::Measurement());
-                        b.measurements.back().rBCc       = cartesianToSpherical(b.cone.axis * distance);
-                        b.measurements.back().covariance = config.ball_angular_cov;
+
+                        // Spherical Coordinates (1/distance, phi, theta)
+                        b.measurements.back().rBCc       = inverseDistanceCartesianToSpherical(b.cone.axis * distance);
+                        b.measurements.back().covariance = config.ball_angular_cov.asDiagonal();
```

Comment:
What does this do?

Reply:
We have taken a measurement of the ball in vision. The measurement is a cone with vertex at the camera and the circular base at the ball.

From this measurement we can determine the distance to the ball from the camera. This is `rBCc` (`camera` to `ball` in `camera` space). However, localisation wants its measurements in spherical coordinates with inverse distance. 

The `covariance` is a measure of how confident we are in the `rBCc` measurement

Reply:
Maybe clarify this in comments, specifically that localisation requires spherical coordinates.